test/city-council-district/city-council-district.e2e-spec.ts

```diff
@@ -104,4 +107,107 @@ describe("City Council District e2e", () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects`)
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(20);
+      expect(parsedBody.offset).toBe(0);
+    });
+
+    it("should 200 amd return all capital projects for a city council district with user specifyied offset and limit", async () => {
+      const limit = 10;
+      const offset = 3;
+      const response = await request(app.getHttpServer())
+        .get(
+          `/city-council-districts/${mock.id}/capital-projects?limit=${limit}&offset=${offset}`,
+        )
+        .expect(200);
+      expect(() =>
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        ),
+      ).not.toThrow();
+      const parsedBody =
+        findCapitalProjectsByCityCouncilIdQueryResponseSchema.parse(
+          response.body,
+        );
+      expect(parsedBody.limit).toBe(limit);
+      expect(parsedBody.offset).toBe(offset);
+    });
+
+    it("should 400 and when finding by an invalid id", async () => {
+      const invalidId = "MN";
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${invalidId}/capital-projects`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an invalid limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=b4d"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an 'too-high' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=25"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+    it("should 400 and when finding by an 'too-low' limit", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?limit=0"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 400 and when finding by an invalid offset", async () => {
+      const response = await request(app.getHttpServer())
+        .get(`/city-council-districts/${mock.id}/capital-projects?offset=b4d"`)
+        .expect(400);
+      expect(response.body.error).toBe(HttpName.BAD_REQUEST);
+    });
+
+    it("should 404 and when finding by an invalid id", async () => {
```

Comment:
In the description, it's more accurate to say "missing" instead of "invalid". 102 is technically valid and could be added to the database under the current structure. It just isn't in there now.